app/controllers/contacts_controller.rb

```diff
@@ -1,13 +1,25 @@
 # frozen_string_literal: true
 
 class ContactsController < ApplicationController
+  rescue_from GdsApi::BaseError do |e|
+    GovukError.notify(e)
+    render "index_api_down", layout: rendering_context
+  end
+
   def index
     @edition = Edition.find_current(document: params[:document])
     assert_edition_state(@edition, &:editable?)
-    @contacts_by_organisation = ContactsService.new.all_by_organisation
     render layout: rendering_context
-  rescue GdsApi::BaseError => e
-    GovukError.notify(e)
-    render "index_api_down", layout: rendering_context
+  end
+
+  def insert
+    if rendering_context == "modal"
+      render inline: I18n.t!("contacts.index.contact_markdown",
+                             id: params[:contact_id])
+    else
+      @edition = Edition.find_current(document: params[:document])
+      assert_edition_state(@edition, &:editable?)
+      render "index", layout: rendering_context
```

Comment:
Rather than repeating this code should we just redirect to index with the contact_id?

Reply:
I wondered about that, but while the no-JS design remains the same, we'll need to use the `insert` action to do the requirements aspect of the selection.